tools/PowerShell/Microsoft.WinGet.Client/src/Library/Find-WinGetPackage.ps1

```diff
@@ -0,0 +1,142 @@
+Function Find-WinGetPackage{
+    <#
+        .SYNOPSIS
+        Searches for a package on configured sources. 
+        Additional options can be provided to filter the output, much like the search command.
+        
+        .DESCRIPTION
+        By running this cmdlet with the required inputs, it will retrieve the packages installed on the local system.
+
+        .PARAMETER Filter
+        Used to search across multiple fields of the package.
+        
+        .PARAMETER Id
+        Used to specify the Id of the package
+
+        .PARAMETER Name
+        Used to specify the Name of the package
+
+        .PARAMETER Moniker
+        Used to specify the Moniker of the package
+
+        .PARAMETER Tag
+        Used to specify the Tag of the package
+        
+        .PARAMETER Command
+        Used to specify the Command of the package
+        
+        .PARAMETER Exact
+        Used to specify an exact match for any parameters provided. Many of the other parameters may be used for case insensitive substring matches if Exact is not specified.
+
+        .PARAMETER Source
+        Name of the Windows Package Manager private source. Can be identified by running: "Get-WinGetSource" and using the source Name
+
+        .PARAMETER Count
+        Used to specify the maximum number of packages to return
+
+        .PARAMETER Header
+        Used to specify the value to pass as the "Windows-Package-Manager" HTTP header for a REST source.
+
+        .PARAMETER VerboseLog
+        Used to provide verbose logging for the Windows Package Manager.
+        
+        .PARAMETER AcceptSourceAgreement
+        Used to accept any source agreement required for the source.
+
+        .EXAMPLE
+        Find-WinGetPackage -id "Publisher.Package"
+
+        This example searches for a package containing "Publisher.Package" as a valid identifier on all configured sources.
+
+        .EXAMPLE
+        Find-WinGetPackage -id "Publisher.Package" -source "Private"
+
+        This example searches for a package containing "Publisher.Package" as a valid identifier from the source named "Private".
+
+        .EXAMPLE
+        Find-WinGetPackage -Name "Package"
+
+        This example searches for a package containing "Package" as a valid name on all configured sources.
+    #>
+    PARAM(
+        [Parameter(Position=0)] $Filter,
```

Comment:
This is very early. Pipelining won't work well with anything depending on list or a display where we truncate strings in the Terminal until we get JSON output via the COM API.

Reply:
Fair point; I guess I’m thinking like a tester in that someone could do something that would end up being like -
```
Install-WingetPackage -Id @{“Id”=“Google.Chrome”}
```

I know its not something that would be common, but if people try writing custom scripts, what happens if an error like that occurrs? I’m assuming it will be fine as is, but just want to be sure that we are aware of any potential adverse effects